### ISSUE TITLE
Error with Move and Order

### ISSUE THREAD

**Issue by dcsparx** [LordAshes/TaleSpire-BeyondLinkViaChrome#8]
Getting a repeating error in the log about an issue with the Move and Order part of the stats.  We changed it to "Move,Move" and it works fine.

https://i.imgur.com/3wQ7Zj5.png

**Comment by LordAshes** [comment]
It appears that the data you are reading is not in the correct format. Dnd
Beyond has been fiddling with their page and you may need to adjust the
provider to compensate.

Best Regards,
Lord Ashes

**Comment by dcsparx** [comment]
Would you be willing to show/tell me where to change that?

**Comment by LordAshes** [comment]
Of course. The data extractor is in the injection.js file found in the
Chrome Extension folder. If you open it up, you will see a bunch of
functions for extracting data and then near the end of the file, the code
that actually calls those functions and formats the JSON object that is
sent to the server. To make things easier, I made functions like GetStat
and GetModifier that uses the "key" string that you provide it to extract
the desired piece of data. If the website changes a bit, those string may
not longer exist (e.g. may have been renamed) and thus you need to see what
they have been changed to.
You can do that by loading your DND change in DndBeyond and then using the
INSPECT browser option to look at the HTML DOM. Do not use the View Source
option, that will not work. Now in the INSPECT you need to find the section
of HTML code that surrounds your data and update the string "key" in the js
file accordingly.
Once you modify the js file, you will need to go to the extension in Chrome
and press the update button (the circular arrow button) to have Chrome
update to using the latest injection.js file.
The best way to test if your extract is correct is to run the server in
LEARN mode (see docs for more info). Don't run Talespire but instead go to
the BeyondLinkServer.exe folder and then run "BeyondLinkServer.exe 9100
LEARN". The LEARN parameter will tell the server to stay open even if
Talespire is not running.
With the server open, refresh your DndBeyond page and the data extractor
will stat sending data to the BeyondLinkServer. You will see changes being
written to the screen but you can also look at the content.json which
contains the whole JSON that was sent from the Chrome Extension. If you see
empty pieces of data or wrong looking data you know you have an issue with
the extractor.

That is a quick summary. If you need more help I suggest coming over to the
Unofficial Talespire Modding Discord and we can chat there to assist you.

Lord Ashes
